toolz/functoolz.py

```diff
@@ -681,3 +659,172 @@ def __name__(self):
             return '%s_excepting_%s' % (self.f.__name__, exc_name)
         except AttributeError:
             return 'excepting'
+
+
+def is_valid_args(func, args, kwargs):
```

Comment:
I did as you suggested.  Thanks for taking a look at this PR, @llllllllll.  I think this PR is now functionally complete, but could stand to use a little more documentation.